org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/IRenameParticipant.java

```diff
@@ -25,7 +26,22 @@
  */
 public interface IRenameParticipant {
 
-	void doRename(IRenameRequest request, List<TextEdit> edits, CancelChecker cancelChecker);
+	/**
+	 * Creates the list of document changes for the rename operation.
+	 * 
+	 * @since 0.26
+	 * @param request A rename request
+	 * @param documentChanges A list to collect either text document edits or rename operations
+	 * @param cancelChecker Cancel checker
+	 */
+	void doRename(IRenameRequest request, List<Either<TextDocumentEdit, ResourceOperation>> documentChanges, CancelChecker cancelChecker);
```

Comment:
I think we could simplify the API by introducing IRenameResponse (like ICompletionRequest/Response).

`void doRename(IRenameRequest request, IRenameResponse response, CancelChecker cancelChecker);`

IRenameResponse could provide 2 methods:

 * addTextDocumentEdit
 * addTextDocumentEdits
 * addResourceOperation
 * addResourceOperations

Reply:
The IRenameResponse interface is added